File: sanitizers/csharp/AntiDriftSanitizer.cs
using System;
using System.Text.RegularExpressions;

namespace OpenAntiDrift.Sanitizer
{
    /// <summary>
    /// Universal Real-Time Token Sanitizer for C# / .NET Applications.
    /// </summary>
    public class AntiDriftSanitizer
    {
        public string UserName { get; }
        public string UserGender { get; }

        public AntiDriftSanitizer(string userName, string userGender)
        {
            UserName = userName;
            UserGender = userGender?.ToLowerInvariant().Trim() ?? "unspecified";
        }

        public string SanitizeText(string text)
        {
            if (string.IsNullOrWhiteSpace(text)) return text;

            bool isFemale = UserGender.Contains("female") || UserGender.Contains("woman") || UserGender.Contains("she");
            bool isMale = UserGender.Contains("male") || UserGender.Contains("man") || UserGender.Contains("he");

            if (!isFemale && !isMale) return text;

            var result = text;
            if (isFemale)
            {
                result = Regex.Replace(result, $@"\b{Regex.Escape(UserName)}\s+(was|is|looks)\s+he\b", $"{UserName} $1 she", RegexOptions.IgnoreCase);
                result = Regex.Replace(result, $@"\b{Regex.Escape(UserName)}'s\s+his\b", $"{UserName}'s her", RegexOptions.IgnoreCase);
            }
            return result;
        }

        public static (string? thought, string visibleText) ExtractThoughtBlock(string text)
        {
            if (string.IsNullOrWhiteSpace(text)) return (null, "");

            var match = Regex.Match(text, @"<thought>([\s\S]*?)</thought>");
            string? thought = match.Success ? match.Groups[1].Value.Trim() : null;
            string visibleText = Regex.Replace(text, @"<thought>[\s\S]*?</thought>", "").Trim();

            return (thought, visibleText);
        }
    }
}

File: sanitizers/kotlin/AntiDriftSanitizer.kt
package open.antidrift.sanitizer

/**
 * Universal Real-Time Token Sanitizer for Kotlin / Android LLM Applications.
 */
class AntiDriftSanitizer(
    private val userName: String,
    private val userGender: String
) {
    fun sanitizeText(text: String): String {
        if (text.isBlank()) return text
        val uLower = userGender.lowercase().trim()
        val isFemale = uLower.contains("female") || uLower.contains("woman") || uLower.contains("she")
        val isMale = uLower.contains("male") || uLower.contains("man") || uLower.contains("he")

        if (!isFemale && !isMale) return text

        var result = text
        if (isFemale) {
            result = result.replace(Regex("\\b${Regex.escape(userName)}\\s+(was|is|looks)\\s+he\\b", RegexOption.IGNORE_CASE), "$userName $1 she")
            result = result.replace(Regex("\\b${Regex.escape(userName)}'s\\s+his\\b", RegexOption.IGNORE_CASE), "$userName's her")
        }
        return result
    }

    companion object {
        fun extractThoughtBlock(text: String): Pair<String?, String> {
            val match = Regex("<thought>([\\s\\S]*?)</thought>").find(text)
            val thought = match?.groupValues?.get(1)?.trim()
            val visibleText = text.replace(Regex("<thought>[\\s\\S]*?</thought>"), "").trim()
            return Pair(thought, visibleText)
        }
    }
}
